File: Chapter10/Python/Using Inheritance to Create a Derived Class in Python/Motorcycle.py
class Motorcycle(object):
    def __init__(self, maxSpeed, speed, sidecar):
        self.sidecar = sidecar
        self.maxSpeed = maxSpeed
        self.speed = speed

    def accelerate(self, accelerate):
        self.accelerate = accelerate
        if (self.speed + self.accelerate) > self.maxSpeed:
            print("This motorcycle cannot go that fast")
        else:
            self.speed += self.accelerate

File: Chapter10/C++/Using Inheritance to Create a Derived Class in C++/Motorcycle.cpp
// Motorcycle.cpp
#include "Vehicle.cpp"
#include <iostream>
using namespace std;

// Write the Motorcycle class here

class Motorcycle : public Vehicle
{
public:
    void setSideCar(bool);
    bool getSideCar();
    //void accelerate(double);

private:
    bool sideCar;
    //double  maxSpeed;
    //double  fuelCapacity;
    //double  currentSpeed;
};
bool Motorcycle::getSideCar()
{
    return sideCar;
}
void Motorcycle::setSideCar(bool sCar)
{
    sideCar = sCar;
    return;
}

// void Motorcycle::accelerate(double mph)
// {
//     if ((currentSpeed + mph) <= maxSpeed)
//         currentSpeed = currentSpeed + mph;
//     else
//         cout << "This motorcycle cannot go that fast." << endl;
// }
